distribution/tools/entitlement-agent/src/main/java/org/elasticsearch/entitlement/agent/EntitlementAgent.java

```diff
@@ -9,13 +9,51 @@
 
 package org.elasticsearch.entitlement.agent;
 
-import org.elasticsearch.entitlement.runtime.api.EntitlementChecks;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.entitlement.checks.EntitlementChecks;
+import org.elasticsearch.entitlement.instrumentation.Instrumenter;
 
+import java.io.File;
+import java.io.IOException;
 import java.lang.instrument.Instrumentation;
+import java.util.List;
+import java.util.jar.JarFile;
+
+import static java.util.stream.Collectors.toSet;
 
 public class EntitlementAgent {
 
     public static void premain(String agentArgs, Instrumentation inst) throws Exception {
-        EntitlementChecks.setAgentBooted();
+        // Add the trampoline library (the one with the entitlement checking interface) to the bootstrap classpath.
+        // We can't actually reference the classes here for real before this point because they won't resolve.
+        var jarsString = System.getProperty("entitlements.trampolineJars");
+        if (jarsString != null) {
+            addJarsToBootstrapClassLoader(inst, jarsString);
+        }
+
+        ConfigurationScanner.ScanResults config = ConfigurationScanner.scan(List.of(EntitlementChecks.class));
```

Comment:
Something needs renaming here -- ScanResults is not very clear, and neither is config (and they do not "match")